app.js

```diff
@@ -119,10 +122,44 @@ app.use(function(req, res, next) {
 });
 
 /**
- * Last chance: is this a static asset?
+ * Is this a static asset?
  */
 app.use(express.static(DIST_DIR));
 
+
+/**
+ * Last chance: check if it's a WP page, otherwise send 404
+ */
+app.use(function(req, res, next) {
+  require('babel-register')({
+    extensions: [".jsx", ".js"],
+    presets: ['react']
+  });
+  var ReactDOMServer = require('react-dom/server');
+  var Page = require('./components/page.jsx');
+  var HtmlWrapper = require('./components/HTML-wrapper.jsx');
+  var WpContentLoader = require('react-wp-content-loader');
+
+  wpPageChecker(req.path, function(err, wpContent) {
+    console.log("\n\n\n\n Page", Page);
+
+    if ( err ) {
+      res.status(404).send(notFoundHTML);
+    } else {
+      var PageContent = React.createElement(Page, 
+        {
+          routes: [
+            { path: '', component: { pageClassName: '', pageTitle: ''} }
+          ] 
+        }, 
+        React.createElement('div', { dangerouslySetInnerHTML: { __html: wpContent }})
+      );
+
+      res.status(200).send("<!DOCTYPE html>" + ReactDOMServer.renderToStaticMarkup(PageContent));
```

Comment:
Updated discovery -

It's related some path issue, namely if you have a `file.extension` and a directory called `file`, things may not work all that well.

In this case `Sidebar` was module exported via `require('./sidebar')`